### PR TITLE
Mm 47648 check license expandable

### PR DESCRIPTION
**Pull request by Afsoon**
#### Summary

Implement an endpoint to check if a license is suitable to be self-expand.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-47648

#### Release Note

```release-note
Endpoint to portal to detect if a license is suitable to self-expand.
```

### CODE REVIEW

File: utils/license.go
Comment (Afsoon) [inline utils/license.go:26]: I have to revert this before the merge. This is only for testing purposes in webapp.

File: api4/cloud_test.go
Comment (neallred) [inline api4/cloud_test.go:706]: Commenting on this PR only because its related to a comment on the webapp side that I'm reviewing. Looks like the test name for this test and the one above uses `expendable` when I think we want `expandable`.

**Review by AGMETEOR — APPROVED** [review]

**Review by marianunez — APPROVED** [review]
LGTM! Thanks @Afsoon!

Is there a related Enterprise PR for this new endpoint?

**Comment by Afsoon** [comment]
> LGTM! Thanks @Afsoon!
> 
> Is there a related Enterprise PR for this new endpoint?

Yes, there is an Enterprise PR for this https://github.com/mattermost/enterprise/pull/1310

File: model/cloud.go
Comment (neallred) [inline model/cloud.go:128]: Nit: I think we want all the code in in webapp, server and enterprise to be `ExpandStatus` instead of `ExpandStats`. I'd expect "stats" to be things like numerical counts of things, and customer-web-server uses "status"

File: api4/cloud_test.go
Comment (neallred) [inline api4/cloud_test.go:675]: Looks like the test name up above and this modal method could use a rename to `Status`

**Review by neallred — APPROVED** [review]
LGTM, thanks @Afsoon !

**Comment by amyblais** [comment]
@marianunez This has a v7.7 milestone. Is this needed for v7.7 or can this go to the next release?

**Comment by marianunez** [comment]
> @marianunez This has a v7.7 milestone. Is this needed for v7.7 or can this go to the next release?

If it's not yet in the v7.7 release branch, it can wait for the next release cc/ @johndavidlugtu

**Comment by marianunez** [comment]
@amyblais could this go into the v7.8 release?

**Comment by amyblais** [comment]
@marianunez Yes 👍

**Pull request merged**
